packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -818,64 +818,67 @@ class HotRunner extends ResidentRunner {
     if (!updatedDevFS.success) {
       return OperationResult(1, 'DevFS synchronization failed');
     }
-    String reloadMessage;
-    final Stopwatch vmReloadTimer = Stopwatch()..start();
+    String reloadMessage = 'Reloaded 0 libraries';
     Map<String, Object> firstReloadDetails;
-    const String entryPath = 'main.dart.incremental.dill';
-    final List<Future<DeviceReloadReport>> allReportsFutures = <Future<DeviceReloadReport>>[];
+    if (updatedDevFS.invalidatedSourcesCount > 0) {
```

Comment:
It might be a good cleanup to factor out the stuff under this if into a helper function.